core/src/test/scala/org/apache/spark/SparkConfSuite.scala

```diff
@@ -140,13 +140,6 @@ class SparkConfSuite extends SparkFunSuite with LocalSparkContext with ResetSyst
     assert(sc.appName === "My other app")
   }
 
-  test("creating SparkContext with cpus per tasks bigger than cores per executors") {
```

Comment:
Is it possible to at least add back this test, for SparkContext?

Reply:
@liutang123 let's address this or close it; some of this is already checked

Reply:
Hi, @srowen I add a UT in SparkContextSuite witch is more reasonable than this I think.
If you think #23290 is necessary to be retained, this PR only requires checking the local mode.
But, imagine this case:
`spark.executor.cores 1` in conf/spark-defaults.conf as default conf.
If user exec `spark-shell --master local[6] --conf spark.task.cpus=2` command, the checking logic in #23290 will throw a exception. This logic forces the user to set `spark.executor.cores` larger than 2 although the `spark.executor.cores` is meaningless. So, I think we can check `spark.task.cpus` before creating SchedulerBackend and TaskScheduler - but what do you think？